github/VisualStudio
# Ensure GitHub.TeamFoundation.15 MEF assembly is always installed on Visual Studio 2017

### What's the issue

Specifying a `TargetVersion` was causing the `GitHub.TeamFoundation.15` MEF assembly to sometimes incorrectly be excluded when installed on Visual Studio 2017 (see #1859).

@dgriffen [said](https://gitter.im/Microsoft/extendvs?at=5b7af3411d3a5711b673493f):
> the bug is that sometimes servicehub's version is used which is not tied directly to the VS version (currently, its ~1.3 I think).

### What this PR does

Assuming this issue is fixed in 2019, we can assume that when the version is reported as `1.x` (instead of `15.x`) we're installing for Visual Studio 2017. Specifying a version range isn't supported when installing for Visual Studio 2015, so this isn't an issue there.

- Install for Visual Studio 2017 when version is incorrectly reported as 1.x

### How to repro

1. Install GitHub for Visual Studio v2.5.4.3349 in Visual Studio 2017
2. Launch Visual Studio 2017 (with the `GitHub` and `Team Explorer` panes closed)
3. Open the folder at: `%USERPROFILE%\AppData\Local\Microsoft\VisualStudio\15.0_xxxxxxxx`
(where `15.0_xxxxxxxx` is a unique name corresponding to the instance of Visual Studio you're running)
4. Delete the folder `ComponentModelCache`
5. Delete the file `Extensions\extensions.en-US.cache`
6. Kill the process `ServiceHub.VSDetouredHost.exe`
![image](https://user-images.githubusercontent.com/11719160/44573637-12a09880-a77f-11e8-978c-bb18a7963b50.png)
7. Open any solution that has a Git repository (you *might* see the dialog below at this point)
8. Close and re-open Visual Studio 2017
9. Open any solution that has a Git repository
10. Expect to see the following dialog:
![image](https://user-images.githubusercontent.com/11719160/44573903-d1f54f00-a77f-11e8-93d6-295b4ec86d21.png)
11. If you've ever accidentally selected no on the above dialog, you won't see anything. 😢 Try running the `View > Other Windows > GitHub`. Expect nothing to happen.

The extensions cache (`extensions.en-US.cache`) is now corrupted. 🎉 😭  The only way to fix it is to delete `Extensions\extensions.en-US.cache` and `ComponentModelCache` and relaunch Visual Studio.

### How to test

Following on from `How to repro`...

1. Install the version from this PR
2. The extension should now be working
3. Repeat from step 2. of `How to repro`
4. Expect the extension to continue working this time

Phew! 🤞 

### Related

Fixes #1859 Corrupt extension cache and missing MEF exports
Fixes #1826 GitHub Extension broken with VS 15.7.6 
Fixes  #1810 SetSite failed for the package [GitHubPackage], VS 15.8 Preview 4/5 
Fixes  #1812 Error On Project Load
